src/animation_commands.c

```diff
@@ -0,0 +1,151 @@
+#include "global.h"
+#include "m4a.h"
+
+#include "data.h"
+#include "flags.h"
+#include "sprite.h"
+
+typedef int (*AnimationCommandFunc)(s32 *variantCursor,
+                                    struct UNK_0808B3FC_UNK240 *sprite);
+
+extern int animCmd_GetTileIndex(s32 *variantCursor, struct UNK_0808B3FC_UNK240 *sprite);
```

Comment:
I think `variantCursor` should actually just be some structure and be called `instruction`. The function which actually calls these depending on the instruction type should either cast the instruction to the type before passing in, or the instructions should all be `void*` so this cast is implicit

Reply:
Another option is `instruction` is a union and each function just does `.whatever` to match the shape its expecting